crates/pallet-humanode-session/src/migrations/v1.rs

```diff
@@ -64,14 +64,21 @@ pub fn migrate<T: Config>() -> Weight {
 /// Panics if anything goes wrong.
 #[cfg(feature = "try-runtime")]
 pub fn pre_migrate<T: Config>() -> Vec<u8> {
-    // Ensure the new identities don't exist yet (i.e. we have clear space to migrate).
-    assert_eq!(<SessionIdentities<T>>::iter().next(), None);
-
-    // Record the count of identities.
-    let identities_count: u64 = <CurrentSessionIdentities<T>>::iter()
-        .count()
-        .try_into()
-        .unwrap();
+    let onchain = <Pallet<T>>::on_chain_storage_version();
+
+    if onchain < 1 {
+        // Ensure the new identities don't exist yet (i.e. we have clear space to migrate).
+        assert_eq!(<SessionIdentities<T>>::iter().next(), None);
+
+        // Record the count of identities.
+        let identities_count: u64 = <CurrentSessionIdentities<T>>::iter()
+            .count()
+            .try_into()
+            .unwrap();
+        return identities_count.encode();
+    }
+
+    let identities_count: u64 = <SessionIdentities<T>>::iter().count().try_into().unwrap();
     identities_count.encode()
```

Comment:
Let's be more explicit here: disable the check and data population entirely for newer versions. Otherwise we might get annoyed by this again in the future.

Reply:
https://github.com/humanode-network/humanode/pull/1005/commits/6b4afa33e59239e9214cc538e0da3801aee67b82

Reply:
At the `post_migrate` check, run the assertions depending on whether the `state` is empty or not. At the `pre_migrate` - only write the `state` if needed. Also, we should make sure the state is non-empty when we encode the `0` identities could - to distinguish between the empty state indicating no checks and the migration of 0 entries.